instrumentation/spring/spring-webmvc-3.1/javaagent/src/main/java/org/springframework/web/servlet/OpenTelemetryHandlerMappingFilter.java

```diff
@@ -90,31 +116,16 @@ private boolean findMapping(HttpServletRequest request) {
   public void setHandlerMappings(List<HandlerMapping> mappings) {
     List<HandlerMapping> handlerMappings = new ArrayList<>();
     for (HandlerMapping mapping : mappings) {
-      // it may be enticing to add all HandlerMapping classes here, but DO NOT
-      //
-      // because we call getHandler() on them above, at the very beginning of the request
-      // and this can be a very invasive call with application-crashing side-effects
-      //
-      // for example: org.grails.web.mapping.mvc.UrlMappingsHandlerMapping.getHandler()
-      // 1. uses GrailsWebRequest.lookup() to get GrailsWebRequest bound to thread local
-      // 2. and populates the servlet request attribute "org.grails.url.match.info"
-      //    with GrailsControllerUrlMappingInfo
-      //
-      // which causes big problems if GrailsWebRequest thread local is leaked from prior request
-      // (which has been observed to happen in Grails 3.0.17 at least), because then our call to
-      // UrlMappingsHandlerMapping.getHandler() at the very beginning of the request:
-      // 1. GrailsWebRequest.lookup() gets the leaked GrailsWebRequest
-      // 2. servlet request attribute "org.grails.url.match.info" is populated based on this leaked
-      //    GrailsWebRequest (so in other words, most likely the wrong route is matched)
-      //
-      // and then GrailsWebRequestFilter creates a new GrailsWebRequest and binds it to the thread
-      //
-      // and then the application calls UrlMappingsHandlerMapping.getHandler() to route the request
-      // but it finds servlet request attribute "org.grails.url.match.info" already populated (by
-      // above) and so it short cuts the matching process and uses the wrong route that the agent
-      // populated caused to be populated into the request attribute above
+      // Originally we ran findMapping at the very beginning of the request. This turned out to have
+      // application-crashing side-effects with grails. That is why we don't add all HandlerMapping
+      // classes here. Although now that we run findMapping after the request, and only when server
+      // span name has not been updated by a controller, the probability of bad side-effects is much
+      // reduced.
```

Comment:
just confirming my understanding (missed this advantage to moving it to the end 👍)
```suggestion
      // reduced even if we did add all HandlerMapping classes here
```